### PR TITLE
Trigger autocomplete with empty input

### PR DESCRIPTION
**Pull request by AkkarinForest**
Also sorts the tags alphabetically, but I added a bit more logic there so tags that have numbers or some weird characters are at the end.

### CODE REVIEW

File: app/javascript/javascript/filtersUtils.js [hunk 1]
```diff
@@ -74,21 +74,44 @@ export const setupTimeRangeFilter = (
...
+          };
+          $.getJSON(path, data, response);
+        },
+        select: function(event, ui) {
+          callback(ui.item.value);
+        },
+        minLength: 0
+      })
+      .focus(function() {
+        $(this).autocomplete("search");
```
Comment (AkkarinForest) [inline app/javascript/javascript/filtersUtils.js]: this two line trigger autocomplete before you start typing

File: app/javascript/javascript/filtersUtils.js [hunk 2]
```diff
@@ -74,21 +74,44 @@ export const setupTimeRangeFilter = (
...
+      });
+  } else {
+    window.setTimeout(setupAutocomplete(callback, path), 100);
+  }
+};
+
+export const setupProfileNamesAutocomplete = callback => {
+  if (document.getElementById("profile-names")) {
+    $(`#profile-names`).autocomplete({
```
Comment (gitjul) [inline app/javascript/javascript/filtersUtils.js]: how about using `document.getElementById("profile-names")` here as well, possibly keeping it in a variable above the if?

File: app/models/api/mobile_tags_params.rb [hunk 3]
```diff
@@ -8,7 +8,7 @@ module Types
 
   Schema =
     Dry::Validation.Schema do
-      required(:input).filled(:str?)
+      required(:input)
```
Comment (gitjul) [inline app/models/api/mobile_tags_params.rb]: is it still required though? what's the value when it's not filled with a string?

File: app/javascript/javascript/filtersUtils.js [hunk 4]
```diff
@@ -74,21 +74,44 @@ export const setupTimeRangeFilter = (
...
-        $.getJSON(path, data, response);
+        $.getJSON("api/autocomplete/usernames", data, response);
       },
       select: function(event, ui) {
         callback(ui.item.value);
       }
     });
   } else {
-    window.setTimeout(setupAutocomplete(callback, id, path), 100);
+    window.setTimeout(setupAutocomplete(callback), 100);
```
Comment (gitjul) [inline app/javascript/javascript/filtersUtils.js]: hmm where's the definition of this function? sorry if I'm blind

File: app/javascript/javascript/filtersUtils.js (hunk 2 above)
Comment (gitjul) [reply app/javascript/javascript/filtersUtils.js]: then we could get rid of jquery in one more place :)

File: app/javascript/javascript/filtersUtils.js [hunk 5]
```diff
@@ -74,21 +74,44 @@ export const setupTimeRangeFilter = (
...
+export const setupProfileNamesAutocomplete = callback => {
+  if (document.getElementById("profile-names")) {
+    $(`#profile-names`).autocomplete({
       source: function(request, response) {
         const data = {
-          q: { input: request.term, ...createParams() }
+          q: { input: request.term }
         };
-        $.getJSON(path, data, response);
+        $.getJSON("api/autocomplete/usernames", data, response);
```
Comment (gitjul) [inline app/javascript/javascript/filtersUtils.js]: would be great to switch to fetch (whatwg-fetch) which we're already using in other places, e.g. `elm.js`
https://github.com/github/fetch

this is a cool source when you suspect you don't need jquery somewhere: http://youmightnotneedjquery.com/

File: app/javascript/javascript/filtersUtils.js (hunk 2 above)
Comment (AkkarinForest) [reply app/javascript/javascript/filtersUtils.js]: Doesn't work. This library only works with elements found by jquery. Cause `Uncaught TypeError: document.getElementById(...).autocomplete is not a function`

File: app/models/api/mobile_tags_params.rb (hunk 3 above)
Comment (AkkarinForest) [reply app/models/api/mobile_tags_params.rb]: It means you can pass `"input":""`, but the keyword input has to be in the params. It also means that in the code later I can do `form[:input]` without  checking if it exists.

File: app/javascript/javascript/filtersUtils.js (hunk 4 above)
Comment (AkkarinForest) [reply app/javascript/javascript/filtersUtils.js]: nowhere 😱  good catch!

File: app/javascript/javascript/filtersUtils.js (hunk 5 above)
Comment (AkkarinForest) [reply app/javascript/javascript/filtersUtils.js]: Again, not sure if it's possible with this jquery library. But we can try get rid of it totally :)

File: app/javascript/javascript/filtersUtils.js (hunk 2 above)
Comment (gitjul) [reply app/javascript/javascript/filtersUtils.js]: 😿

**Pull request merged**